apps/financial-aid/web-veita/src/components/AdminLayout/AdminLayout.tsx

```diff
@@ -15,13 +16,38 @@ const AdminLayout = ({ children, className }: PageProps) => {
     document.title = 'Veita • Umsóknir um fjárhagsaðstoð'
   }, [])
 
+  const router = useRouter()
+  const [showNavMobile, setShowNavMobile] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (showNavMobile) {
+      router.events.on('routeChangeComplete', () => {
+        setShowNavMobile(false)
+      })
+    }
+  }, [showNavMobile])
+
   return (
     <>
-      <Nav />
+      <Nav showInMobile={showNavMobile} />
       <div className={` wrapper ${styles.gridWrapper} `}>
+        <button
+          className={cn({
+            [`${styles.burgerMenu} burgerMenu`]: true,
+            [`openBurgerMenu`]: showNavMobile,
+          })}
+          onClick={() => {
+            setShowNavMobile(!showNavMobile)
```

Comment:
Here the state depends on the previous state so instead of doing this it is better practice todo: 
`setShowNavMobile(showNavMobile => !showNavMobile)`